test/workbox-window/window/test-Workbox.mjs

```diff
@@ -289,15 +289,35 @@ describe(`[workbox-window] Workbox`, function() {
   });
 
   describe(`update`, function() {
-    it(`updates service worker and resolves with the new registration`, async function() {
-      const controllerBeforeTest = navigator.serviceWorker.controller;
-      const scriptURL = controllerBeforeTest.scriptURL;
+    it(`calls update on the registration`, async function() {
+      const scriptURL = navigator.serviceWorker.controller.scriptURL;
       const wb = new Workbox(scriptURL);
+      const reg = await wb.register();
+
+      sandbox.stub(reg, 'update');
 
+      await wb.update();
+
+      expect(reg.update.callCount).to.equal(1);
+    });
+
+    it(`triggers an updatefound event if the SW was udpated`, async function() {
+      const scriptURL = navigator.serviceWorker.controller.scriptURL;
+
+      const wb = new Workbox(scriptURL);
       const reg = await wb.register();
-      const newReg = await wb.update();
+      await wb.controlling;
+
+      // Update the SW after so an update check triggers an update.
+      await updateVersion('2.0.0', scriptURL);
 
-      expect(reg).not.to.equal(newReg);
+      const updatefoundSpy = sandbox.spy();
+      reg.addEventListener('updatefound', updatefoundSpy);
+
+      await wb.update();
+      await waitUntil(() => updatefoundSpy.callCount === 1);
+
+      expect(reg.installing).to.not.equal(navigator.serviceWorker.controller);
```

Comment:
Is there a race condition here, where `reg.wating` (or `reg.active`; not sure if `skipWaiting()` is being called) might be set instead of `reg.installing` if the updated service worker's `install` handler finishes quickly?

I would feel better if there were an assertion making sure that `reg.installing` is not `null`.

Or alternatively, it looks like you work around this elsewhere with

```
reg.addEventListener('updatefound', () => {
  updatedSW = reg.installing;
});
```

Reply:
How about I make the assertion in the `updatefound` event listener?

Reply:
I am pretty sure that when `updatefound` is triggered, `reg.installing` will be set: https://w3c.github.io/ServiceWorker/#service-worker-registration-updatefound-event

So stashing away a reference via `updatedSW = reg.installing` should be fine, and then you could change your assertion in this line to

```
expect(updatedSW).to.not.equal(navigator.serviceWorker.controller);
```

Reply:
Updated. PTAL.